src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/geoMap.js

```diff
@@ -249,128 +261,118 @@ class GeoMapVisualization extends Visualization {
+    centerPoint() {
+        const { x, y } = calculateCenterPoint(this.dataPoints)
+        return { latitude: x, longitude: y }
     }
 
     /**
-     * Helper for prepareDataPoints, pushes `GEO_POINT`'s to the list, and calculates central point.
-     * @returns {{latitude: number, longitude: number}} - center.
+     * Sets size of the visualization.
+     * @param size - new size, list of two numbers containing width and height respectively.
      */
-    calculateExtremesAndPushPoints(
-        dataPoints,
-        preparedDataPoints,
-        accentColor
-    ) {
-        let latitudes = []
-        let longitudes = []
-        dataPoints.forEach((e) => {
-            if (e.type === GEO_POINT) {
-                this.pushGeoPoint(preparedDataPoints, e, accentColor)
-                latitudes.push(e.latitude)
-                longitudes.push(e.longitude)
+    setSize(size) {
+        this.dom.setAttributeNS(null, 'width', size[0])
+        this.dom.setAttributeNS(null, 'height', size[1])
+        this.mapElem.setAttributeNS(
+            null,
+            'style',
+            'width:' + size[0] + 'px;height: ' + size[1] + 'px;'
+        )
+    }
+}
+
+/**
+ * Extract the visualisation data from a full configuration object.
+ */
+function extractFromFullConfig(parsedData, preparedDataPoints, accentColor) {
+    if (parsedData.type === SCATTERPLOT_LAYER && parsedData.data.length) {
+        pushPoints(parsedData.data, preparedDataPoints, accentColor)
+    } else if (parsedData.type === GEO_MAP && ok(parsedData.layers)) {
+        parsedData.layers.forEach((layer) => {
+            if (layer.type === SCATTERPLOT_LAYER) {
+                let dataPoints = layer.data || []
+                pushPoints(dataPoints, preparedDataPoints, accentColor)
+            } else {
+                console.warn('Geo_Map: Currently unsupported deck.gl layer.')
             }
         })
-        let latitude = 0.0
-        let longitude = 0.0
-        if (latitudes.length && longitudes.length) {
-            let minLat = Math.min.apply(null, latitudes)
-            let maxLat = Math.max.apply(null, latitudes)
-            latitude = (minLat + maxLat) / 2
-            let minLon = Math.min.apply(null, longitudes)
-            let maxLon = Math.max.apply(null, longitudes)
-            longitude = (minLon + maxLon) / 2
-        }
-        return { latitude, longitude }
     }
+}
 
-    /**
-     * Pushes a new deck.gl-compatible point made out of `GEO_POINT`
-     *
-     * @param preparedDataPoints - List holding geoPoints to push the new element into.
-     * @param geoPoint           - `GEO_POINT` to create new deck.gl point from.
-     * @param accentColor        - accent color of IDE if `GEO_POINT` doesn't specify one.
-     */
-    pushGeoPoint(preparedDataPoints, geoPoint, accentColor) {
+/**
+ * Extract the visualisation data from a dataframe.
+ */
+function extractFromDataFrame(parsedData, preparedDataPoints, accentColor) {
```

Comment:
I think the name of the function could be better. The docs tell about extracting VISUALIZATION DATA. Lets add it to the name of the function as the name doesn't tell what we are extracting there.

Reply:
Done.